cot/src/utils/graph.rs

```diff
@@ -4,6 +4,7 @@ use thiserror::Error;
 #[error("Cycle detected in the graph")]
 pub struct CycleDetected;
 
+#[doc(hidden)] // not part of public API
 pub fn apply_permutation<T>(items: &mut [T], order: &mut [usize]) {
```

Comment:
Wouldn't `pub(crate)` be enough?

Reply:
This is used in the Cot CLI, so `pub` is needed. I will update the comment to make this clear.